tableauserverclient/server/request_factory.py

```diff
@@ -142,32 +142,29 @@ def update_req(self, group_item, default_site_role):
 
 
 class PermissionRequest(object):
-    def _add_capability(self, parent_element, capability_set, mode):
-        for capability_item in capability_set:
-            capability_element = ET.SubElement(parent_element, 'capability')
-            capability_element.attrib['name'] = capability_item
-            capability_element.attrib['mode'] = mode
-
-    def add_req(self, permission_item):
+    def add_req(self, item, permission_item):
         xml_request = ET.Element('tsRequest')
         permissions_element = ET.SubElement(xml_request, 'permissions')
+        item_element = ET.SubElement(permissions_element, type(item).__name__.rstrip('Item').lower())
```

Comment:
@t8y8 this little bit is terrible, looking for feedback here.

Trying to get the item type (`Workbook`, `Datasource`, `Project`) without having to explicitly send it every time.  You had previously mentioned you want the `Permissions` module to be dumb to the type of .  Figuring that if the function accepts `item_type`, then we will need to create an Enum for that which goes against your recommendation.

TBH it might be nice having each of the `WorkbookItem`, `DatasourceItem`, etc... to have an enum/type string associated with them so you could just use `item.type` instead of this.

Reply:
I was thinking about this over the weekend. Given no changes, I'd prefer explicit 'isinstance' checks because at least we're not relying on the name but that's not much better.

I think including a `.type` str or enum on the ModelItems is better. Maybe call it `_xml_type` if just to be extra conservative.